SofaKernel/framework/sofa/defaulttype/Mat.h

```diff
@@ -599,10 +604,26 @@ class Mat : public helper::fixed_array<VecNoInit<C,real>,L>
             this->elems[i]-=m[i];
     }
 
+
+		/// invert this
+		Mat<L,C,real>& inverted()
+		{
+			Mat<L,C,real> m = *this;
+			invertMatrix(*this, m);
+			return *this;
+		}
+
     /// Invert matrix m
     bool invert(const Mat<L,C,real>& m)
     {
-        return invertMatrix(*this, m);
+				if (&m == this)
```

Comment:
You probably need to `assert( L == C && "cannot invert a non-square matrix");` at some point.

Or even better, only define `invert` when the matrix is square.